frontend/src/modules/WellLogViewer/view/SubsurfaceLogViewerWrapper.tsx

```diff
@@ -0,0 +1,265 @@
+import { WellPicksLayerData } from "@modules/Intersection/utils/layers/WellpicksLayer";
+import { WellLogViewer } from "@webviz/well-log-viewer";
+import { Info } from "@webviz/well-log-viewer/dist/components/InfoTypes";
+import { TemplateTrack } from "@webviz/well-log-viewer/dist/components/WellLogTemplateTypes";
+import { WellLogController } from "@webviz/well-log-viewer/dist/components/WellLogView";
+
+import { isEqual } from "lodash";
+
+import { ReadoutWrapper } from "./ReadoutWrapper";
+
+import { InterfaceTypes } from "../interfaces";
+import { createLogTemplate } from "../utils/logViewerTemplate";
+import { createLogViewerWellpicks, createWellLog } from "../utils/queryDataTransform";
+
+const AXIS_MNEMOS = {
+    md: ["RKB", "DEPTH", "DEPT", "MD", "TDEP", "MD_RKB"],
+    tvd: ["MSL", "TVD", "TVDSS", "DVER", "TVD_MSL"],
+    time: ["TIME"],
+};
+
+const AXIS_TITLES = {
+    md: "MD",
+    tvd: "TVD",
+    time: "TIME",
+};
+
+type GlobalHoverMd = GlobalTopicDefinitions["global.hoverMd"];
+
+export type SubsurfaceLogViewerWrapperProps = {
+    // Data
+    wellboreHeader: WellboreHeader_api | null;
+    curveData: WellboreLogCurveData_api[];
+    trajectoryData: WellboreTrajectory_api;
+    intersectionReferenceSystem: IntersectionReferenceSystem;
+    wellpicks: WellPicksLayerData;
+
+    // Viewer config
+    horizontal: boolean;
+    padDataWithEmptyRows: boolean;
+    templateTracks: TemplateTrack[];
+
+    // Passing the module props to make context and service access less cumbersome
+    moduleProps: ModuleViewProps<InterfaceTypes>;
+};
+
+function useGloballySyncedMd(
+    wellboreHeader: WellboreHeader_api | null,
+    wellLogController: WellLogController | null,
+    workbenchServices: WorkbenchServices,
+    viewContext: ViewContext<InterfaceTypes>
+) {
+    const instanceId = viewContext.getInstanceIdString();
+    const wellboreUuid = wellboreHeader?.wellboreUuid ?? "";
+
+    const lastRecievedChange = React.useRef<GlobalHoverMd>(null);
+
+    const handleGlobalValueChange = React.useCallback(
+        (newValue: GlobalHoverMd) => {
+            if (!isEqual(lastRecievedChange, newValue)) {
+                lastRecievedChange.current = newValue;
+
+                if (newValue?.wellboreUuid === wellboreUuid) {
+                    wellLogController?.selectContent([newValue.md, undefined]);
+                }
+            }
+        },
+        [wellLogController, wellboreUuid]
+    );
+
+    const broadcastGlobalMdChange = React.useCallback(
+        (newMd: number | null) => {
+            const payload: GlobalHoverMd = newMd ? { wellboreUuid, md: newMd } : null;
+
+            workbenchServices.publishGlobalData("global.hoverMd", payload, instanceId);
+        },
+        [instanceId, wellboreUuid, workbenchServices]
+    );
+
+    React.useEffect(() => {
+        return workbenchServices.subscribe("global.hoverMd", handleGlobalValueChange, instanceId);
+    }, [handleGlobalValueChange, instanceId, workbenchServices]);
```

Comment:
Why not handling everything inside the useEffect instead of creating a callback?